tests/conftest.py

```diff
@@ -27,6 +27,13 @@
 PIR_TO_XY_RESPONSES = sorted(PIR_TO_XY_FOLDER.iterdir())
 
 
+@pytest.fixture(autouse=True)
+def custom_cache_dir(monkeypatch, tmpdir):
+    # Automatically use a custom cache directory for all tests to avoid writing
+    # cache data to the user's real cache.
+    monkeypatch.setenv("XDG_CACHE_HOME", str(tmpdir))
```

Comment:
Cool! So since this is function-scoped for each test there will be a separate `cache_dir`, right? Do you think we should worry about scaling since `n_temp_dirs = n_tests`. I would guess that the vast majority of tests do not work with the cache folder so in a way it is very wasteful.

Alternative approaches
1. Write out `monkeypatch.setenv("XDG_CACHE_HOME", str(tmpdir))` in each test where the cache folder is actually used. Of course there is a danger we just forget about it and the real one is used.
2. The same as 1), however, using `monkeypatch.setattr("user_cache_dir", fake_user_cache_dir)` instead. The functionality should be the same, however, we kind of pretend that we do not see what is going on inside of the function.

Reply:
I had the same worry, but it seems the overhead is realy minimal. Running `pytest` locally is still really fast and I didn't notice any slowdown. Sooo if there's barely any slowdown and it solves all our potential issues forever, I would be in favour of sticking with this simple solution. What do you think?

Reply:
Sure, let's keep it the way it is. If it ever causes a slowdown we can revisit it:)